pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -97,56 +97,6 @@ func TestReconcile(t *testing.T) {
 				assert.Equal(t, hivev1.InitializedConditionReason, runCond.Reason)
 			},
 		},
-		{
-			name: "do not hibernate unsupported versions",
-			cd:   cdBuilder.Options(testcd.WithClusterVersion("4.3.11")).Build(),
-			cs:   csBuilder.Build(),
-			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
-				cond, _ := getHibernatingAndRunningConditions(cd)
-				require.NotNil(t, cond)
-				assert.Equal(t, corev1.ConditionFalse, cond.Status)
-				assert.Equal(t, hivev1.HibernatingReasonUnsupported, cond.Reason)
-			},
-		},
-		{
-			name: "set powerstate for unsupported clusters",
-			cd: cdBuilder.Options(testcd.WithClusterVersion("4.3.11"),
-				testcd.WithCondition(hivev1.ClusterDeploymentCondition{
-					Type:    hivev1.ClusterHibernatingCondition,
-					Status:  corev1.ConditionFalse,
-					Reason:  hivev1.HibernatingReasonUnsupported,
-					Message: "Unsupported version, need version 4.4.8 or greater"})).Build(),
-			cs: csBuilder.Build(),
-			setupActuator: func(actuator *mock.MockHibernationActuator) {
-				actuator.EXPECT().MachinesRunning(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(true, nil, nil)
-			},
-			setupRemote: func(builder *remoteclientmock.MockBuilder) {
-				objs := []runtime.Object{}
-				objs = append(objs, readyNodes()...)
-				objs = append(objs, readyClusterOperators()...)
-				c := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(objs...).Build()
-				builder.EXPECT().Build().Times(1).Return(c, nil)
-			},
-			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
-				cond, runCond := getHibernatingAndRunningConditions(cd)
-				require.NotNil(t, cond)
-				assert.Equal(t, hivev1.HibernatingReasonUnsupported, cond.Reason)
-				require.NotNil(t, runCond)
-				assert.Equal(t, hivev1.ReadyReasonRunning, runCond.Reason)
-				assert.Equal(t, hivev1.ClusterPowerStateRunning, cd.Status.PowerState)
-			},
-		},
```

Comment:
These tests actually also cover the "Unsupported" reason path - i.e., it gets set when expected without errors, and powerstate gets updated for unsupported clusters. One criterion of determining the hibernation is unsupported is the old clusterversion, while the other one is that the cloud platform (and hive) should support hibernation. I think we should keep these 2 tests, just alter them to point to clusters with unsupported platforms.
PS: A bit of history - iirc, we included one of these tests because of a bug QEs found during the initial kerfuffle of the previous refactor.